File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini=0;
        int maxp=0;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<prices[mini]){
                mini=i;
            }
            else{
                int p=prices[i]-prices[mini];
                if(p>maxp){
                    maxp=p;
                }
            }
        }
        return maxp;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=prices[0];
        int maxp=0;
        for(int i=0;i<prices.size();i++){
            maxp=max(maxp,prices[i]-mini);
            mini=min(mini,prices[i]);

        }
        return maxp;
        
    }
};